test/lightning_web/live/workflow_live/editor_test.exs

```diff
@@ -261,6 +261,38 @@ defmodule LightningWeb.WorkflowLive.EditorTest do
       assert view |> has_element?("#manual-job-#{job.id} form", "Invalid JSON")
     end
 
+    test "can't run if limit is exceeded", %{
+      conn: conn,
+      project: %{id: project_id},
+      workflow: w
+    } do
+      job = w.jobs |> hd
+
+      {:ok, view, _html} =
+        live(conn, ~p"/projects/#{project_id}/w/#{w}?#{[s: job, m: "expand"]}")
+
+      assert Invocation.list_dataclips_for_job(job) |> Enum.count() == 0
+
+      Mox.stub(
+        Lightning.Extensions.MockUsageLimiter,
+        :limit_action,
+        &Lightning.Extensions.StubUsageLimiter.limit_action/2
+      )
+
+      assert view
+             |> form("#manual-job-#{job.id} form",
+               manual: %{
+                 body: Jason.encode!(%{"a" => 1})
+               }
+             )
+             |> render_submit()
+             |> Floki.parse_fragment!()
+             |> Floki.find("#flash")
+             |> Floki.find("p")
+             |> Floki.text() =~
+               "Runs limit exceeded"
```

Comment:
Btw, you don't have to use Floki for these kinds of assertions (if you don't want to).

You can:

```elixir
view |> form(...) |> render_submit()
assert view |> has_element("#flash p", "Runs limit exceeded")
```

Or if you want to use Floki you can probably combine the two find calls to be `#flash p`.

Not asking a change if you don't want to, just a btw.. :)

Reply:
Thank you